refactor(devtools): guard value reads in the serializer

JeanMeche opened the pull request:
Some properties (like gets) might throw when we try to read them. With this commit we fail gracefuly and show a warning message for the property that can't be read.

<img width="344" height="187" alt="Screenshot 2025-09-26 at 02 08 46" src="https://github.com/user-attachments/assets/792e6de3-c4a7-4a76-9700-463c57a541de" />

fixes #56755

dgp1130 commented on devtools/projects/ng-devtools-backend/src/lib/state-serializer/state-serializer.ts:
```diff
@@ -81,7 +81,20 @@ function levelSerializer(
+    serializableInstance = instance[propName];
+  } catch {
+    return {
+      type: PropType.Unknown,
```
Consider: It's a minor point here, but I feel like this might be mixing some view layer concerns into the backend. From the DevTools panel's perspective, this looks like a regular string.

Should we create some kind of `PropType.Error` in the backend and then let the UI layer convert that into the error text?

JeanMeche replied:
Yeah It's a good suggestion to add an error type.

JeanMeche committed:
refactor(devtools): guard value reads in the serializer

Some properties (like gets) might throw when we try to read them.
With this commit we fail gracefuly and show a warning message for the property that can't be read.

fixes #56755
